### ISSUE TITLE
Consider Supporting Yarn Plug'n'Play

### ISSUE THREAD

**Issue by Kurt-von-Laven** [tmorell/license-compliance#153]
**Is your feature request related to a problem? Please describe.**
[Yarn PnP](https://yarnpkg.com/features/pnp) is an optimized replacement for `node_modules` that most npm packages support without any additional work on their part, but unfortunately because of the nature of `license-compliance`, it appears to be incompatible with Yarn PnP. `yarn run license-compliance` outputs a stream of many messages that say: `Package "<package_name>" was not found. Confirm that all modules are installed.` I ran `yarn install && yarn dedupe --strategy highest` to confirm that all packages were installed to no avail.

**Describe the solution you'd like**
It would be ideal to get the dependencies tree in a PnP-compatible fashion. Yarn has [an API](https://yarnpkg.com/api/classes/yarnpkg_core.workspace.html#dependencies) that may make this doable without much work.

**Describe alternatives you've considered**
Try to find a different tool with similar features that supports Yarn PnP.

**Comment by tmorell** [comment]
@Kurt-von-Laven Thank you  for the suggestion. I will look into it, but I cannot commit to implement it at this moment. Let me know if you are interested in providing a PR or a proof of concept.

**Comment by Kurt-von-Laven** [comment]
Yeah, I am interested in working on this. If I understand correctly, the existing code traverses the `node_modules` directory directly. One option would be to instead use the aforementioned Yarn API whenever `yarn.lock` is present. How does that sound?

**Comment by tmorell** [comment]
Correct, but the current `node_modules` transversal should be abstracted, so it can detect `node_modules` vs. `Yarn Plug'n'Play`.

**Comment by Kurt-von-Laven** [comment]
Looks like I was beat to the punch! See pivotal/LicenseFinder#936. There are still some issues to sort out, because Yarn doesn't locally install dependencies like fsevents on Linux or Windows that are only intended for use on MacOS. If it's okay with you, I will close this issue once a released version of LicenseFinder fully supports Yarn PnP and otherwise return to add Yarn PnP support here.

**Comment by tmorell** [comment]
I'll keep it open. The dependencies traversal needs to be abstracted.
As it is, it is not compatible with `pnpm` either.
